tests_require.txt

```diff
@@ -0,0 +1,9 @@
+pytest>=3.5.0
```

Comment:
Can we rename a file to something else, e.g. test_requirements.txt. Even better to move to the tox and pinned dependencies.

Reply:
I had named it `tests_require` so that tab completion is easier (with the `tests` directory next to it, so it doesn't clash with `test_`). I thought the file name is sufficiently obvious as it is? If you think it's important to call it `requirements` we can rename it. I don't understand what you mean by "move to the tox and pinned dependencies", can you explain please?

Reply:
* The `*requirements.txt` is standard naming for requirements files, let's stick to that name.
* I meant that ideally this is also intermediate solution and we should move to the tox https://tox.readthedocs.io/en/latest/ configuration. Pinned or locked dependency file is another thing which we need - usage of the `latest` versions for deps is quite a bad practice.

Reply:
Ok, I renamed it.

Why would it be bad practice to simply depend on e.g. `tensorflow` ? I thought we'd want to be compatible with latest version of tensorflow, scipy etc - so if a test suddenly fails because it pulled in a new dependency that's the quickest way of finding out & adjusting as needed, I would think. Also, otherwise we force people to use specific versions of the dependencies, doesn't seem very user-friendly?
Could you give some references to articles explaining why that would be bad practice, what would be considered wrong about that?